docs/how-to/profiling.txt

```diff
@@ -0,0 +1,81 @@
+.. _howto-profiling:
+
+#################################
+ How To: Profile Your Experiment
+#################################
+
+When you want to improve the training speed of your model, the first
+thing you need to do is understand why training is slow. To do this, you
+need to profile your training code. There are many different layers you
+can look at, from raw throughput all the way to GPU kernals.
```

Comment:
```suggestion
When you want to debug the training speed of a model, you need
tools to inspect and measure internal states and runtime statistics of
the model during execution. There are many different layers you
can look at, from raw throughput all the way to GPU kernels.
```
Idk this is a hard paragraph to write.. i want to take a course on technical writing. I feel like there is a better way to motivate the reader but i can't quite find the words

Reply:
I rewrote it a bit. It could probably still use some improvements.